contracts/native/governance/node_manager/manager.go

```diff
@@ -247,9 +255,17 @@ func UpdateValidator(s *native.NativeContract) ([]byte, error) {
 
 	if params.ProposalAddress != common.EmptyAddress {
 		validator.ProposalAddress = params.ProposalAddress
+		err = setProposalAddr(s, validator.ProposalAddress)
+		if err != nil {
+			return nil, fmt.Errorf("UpdateValidator, setProposalAddr error: %v", err)
+		}
 	}
 	if params.SignerAddress != common.EmptyAddress {
 		validator.SignerAddress = params.SignerAddress
+		err = setSignerAddr(s, validator.SignerAddress)
+		if err != nil {
+			return nil, fmt.Errorf("UpdateValidator, setSignerAddr error: %v", err)
+		}
 	}
 
```

Comment:
Missing code to remove the stale addresses.

Reply:
done